spec/controllers/holds_controller_spec.rb

```diff
@@ -350,14 +354,38 @@
         get :result, params: {}, session: { place_hold_catkey: '1', place_hold_results: results }
 
         placed_hold = assigns(:place_hold_results)[:success].first[:placed_hold]
-        expect(placed_hold.record.first['key']).to eq 'a_hold_key'
+        expect(placed_hold.record['key']).to eq 'a_hold_key'
       end
 
       it 'assigns failed holds results correctly' do
         get :result, params: {}, session: { place_hold_catkey: '1', place_hold_results: results }
 
         failed_hold = assigns(:place_hold_results)[:error].first[:failed_hold]
-        expect(failed_hold.record.first['fields']['barcode']).to eq 'not_holdable_barcode'
+        expect(failed_hold.record['fields']['barcode']).to eq 'not_holdable_barcode'
+      end
+
+      context 'when hold lookup does not return item level info' do
+        let(:hold_info_empty_item_response) { HOLD_LOOKUP_NIL_ITEM_RAW_JSON.to_json }
+        let(:parsed_hold_info) { JSON.parse hold_info_response }
+        let(:parsed_hold_info_empty_item) { JSON.parse hold_info_empty_item_response }
+        let(:hold_info_params) { ['a_hold_key', user[:session_token]] }
+        let(:mock_client) { instance_double(SymphonyClient) }
+
+        let(:results) { {
+          success: [{ barcode: 'holdable_barcode', hold_key: 'a_hold_key' }]
+        }.with_indifferent_access }
+
+        before do
+          allow(SymphonyClientParser).to receive(:parsed_response).with(mock_client, :get_hold_info, *hold_info_params)
+            .and_return(parsed_hold_info_empty_item, parsed_hold_info)
+        end
+
+        it 'tries hold lookup again until item info returns not empty' do
```

Comment:
It isn't clear to me how this is being tested, sorry. How does this spec prove that a second call to the symphony client is made when items is `nil`? I think I was expecting some kind of (pseudo code) `SymphonyClient calls.count == 2` or something, idk. This seems like it's just showing that the second object passed back from the `SymphonyClientParser` is valid. Sorry again if I'm totally missing this, it's been a while since I've looked at this.

Reply:
```
allow(SymphonyClientParser).to receive(:parsed_response).with(mock_client, :get_hold_info, *hold_info_params)
            .and_return(parsed_hold_info_empty_item, parsed_hold_info)
```
First call to SymphonyClientParser returns the empty item response and the second call returns the response with hold info. So, I did not think we necessarily have to test the empty case or the number of totals calls as long as the final result is the response with hold info. But I think we can add another example to test the calls count if you think it makes it more clear.

Reply:
No we're good, I got it now from our conversation earlier in the issue, thanks for spelling it out 👍